src/GitHub.Api/Authentication/LoginManager.cs

```diff
@@ -119,7 +119,17 @@ class LoginManager : ILoginManager
                         throw new InvalidOperationException("Returned token is null or empty");
                     }
 
-                    keychain.SetToken(host, loginResultData.Token);
+                    var octorunTask = new OctorunTask(taskManager.Token, nodeJsExecutablePath.Value, octorunScript.Value, "validate",
+                            user: username, userToken: loginResultData.Token)
+                        .Configure(processManager);
+
+                    var validateResult = await octorunTask.StartAsAsync();
+                    if (!validateResult.IsSuccess)
+                    {
+                        throw new InvalidOperationException("Authentication validation failed");
+                    }
+
+                    keychain.SetToken(host, loginResultData.Token, validateResult.Output[1]);
```

Comment:
This is not the only place where the username needs to be set, non-2fa login also needs it. And since we're always setting the user in the credential, why not make the username in `SetToken` mandatory? We always have it available in some form, and making it mandatory will allow us to make sure it's always sent in.